File: Data Structures & Algorithms/duplicate-integer/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums: number[]): boolean {
        const freq = new Map<number, number>;
        for (const num of nums) {
            if (freq.get(num)) return true;
            freq.set(num, 1);
        }
        return false;
    }
}

File: Data Structures & Algorithms/majority-element/submission-1.cpp
class Solution {
   public:
    int majorityElement(vector<int>& nums) {
        for (int num: nums) {
            int watch;
            int count=0;
            for (int num: nums) {
                if (!count) {
                    watch = num;
                    count = 1;
                } else if (num!=watch) {
                    count--;
                } else count++;
            }
            return watch;
        }
    }
};